src/hotspot/share/opto/gcm.cpp

```diff
@@ -769,6 +773,40 @@ Block* PhaseCFG::insert_anti_dependences(Block* LCA, Node* load, bool verify) {
     }
   }
   worklist_def_use_mem_states.push(nullptr, initial_mem);
+  Block* initial_mem_block = get_block_for_node(initial_mem);
+  assert(initial_mem_block != nullptr, "sanity");
+  if (load->in(0) != nullptr) {
+    // If the load has an explicit control input, walk up the dominator tree
+    // from the early block (inclusive) to the initial memory block
+    // (inclusive). If we in a block find memory Phi(s) that can alias
+    // initial_mem, these are also potential initial memory states and there
+    // may be further required anti-dependences due to them.
+    assert(initial_mem_block->dominates(early), "invariant");
+    Block* b = early;
+    // Stop searching when we step past the initial memory block (b ==
+    // initial_mem_block->_idom). The loop below always terminates because the
+    // root block strictly dominates initial_mem_block.
+    while (b != initial_mem_block->_idom) {
```

Comment:
Could you write a `for` instead?
`for(Block* b = early; b != initial_mem_block->_idom; b = b->_idom) {`

Having the initial, exit-check and iteration-step together makes it a little more readable, I think.

Reply:
Sure, sounds good. I'll update it.